src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -2266,6 +2296,29 @@ theorem toNat_umod {x y : BitVec n} :
 theorem umod_zero {x : BitVec n} : x % 0#n = x := by
   simp [umod_def]
 
+@[simp]
+theorem zero_umod {x : BitVec w} : (0#w) % x = 0#w := by
+  simp [bv_toNat]
+
+@[simp]
+theorem umod_one {x : BitVec w} : x % (1#w) = 0#w := by
+  simp only [toNat_eq, toNat_umod, toNat_ofNat, Nat.zero_mod]
+  cases w
+  · simp [eq_nil x]
+  · simp [Nat.mod_one]
+
+@[simp]
+theorem umod_self {x : BitVec w} : x % x = 0#w := by
+  simp [bv_toNat]
+
+@[simp]
+theorem umod_eq_and {x y : BitVec 1} : x % y = x &&& (~~~y) := by
+  have hx : x = 0#1 ∨ x = 1#1 := by bv_omega
+  have hy : y = 0#1 ∨ y = 1#1 := by bv_omega
+  rcases hx with rfl | rfl <;>
+    rcases hy with rfl | rfl <;>
+      rfl
```

Comment:
```suggestion
theorem umod_eq_and {x y : BitVec 1} : x % y = x &&& (~~~y) := by revert x y; decide
```

Same trick: finitely many cases, so use the `decide` instance.

Reply:
decideable instances are defined later only.